### PR TITLE
App update + new option to avoid changing the value manually in the autoincrement mode

### PR DESCRIPTION
**Pull request by grzesiek2010**
#### What has been done to verify that this works as intended?
I've tested the changes manually.

#### Why is this the best possible solution? Were any other approaches considered?
We had to update the app so that it works well on newer devices. Additionally, we were asked to extend the functionality by adding the `autoincrement` mode. There were no other approaches to consider.

#### Are there any risks to merging this code? If so, what are they?
Yes, a lot of things have been refactored so the whole app needs to be tested.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,6 +31,8 @@ appearance="ex:org.opendatakit.counter(form_id='counter-form', form_name='Counte
 
 **increment=true()** - an optional parameter. If set to true an initial value displayed in the Counter app is 0 and it will be increased by 1 after 0.5s otherwise the initial value is set to 1 and autoincrement doesn't take place.
 
+**autoincrement=true()** - an optional parameter. If set to true it works like **increment=true()** but additionally the buttons to manually change the value are disabled.
```
Comment (lognaturel) [inline README.md:34]: What happens if both are true? Maybe document that `autoincrement` takes precedence?

**Review by lognaturel — COMMENTED** [review]
I looked through this quite quickly. All the high-level changes look like ones we want to make.

The most important thing is verification! @grzesiek2010 is there more to do here? I feel like beyond the small docs change this can be marked for testing.

File: README.md (hunk 1 above)
Comment (grzesiek2010) [reply README.md:34]: In that case, as you said `autoincrement` takes precedence. I've added it to the `README`.

**Comment by grzesiek2010** [comment]
> I looked through this quite quickly. All the high-level changes look like ones we want to make.

> The most important thing is verification! @grzesiek2010 is there more to do here? I feel like beyond the small docs change this can be marked for testing.

This is finished and contains everything that we were asked to do. I think tomorrow the QA team can work on it.

**Comment by srujner** [comment]
I'll suggest a small improvement for the disabled buttons color;
Disabled buttons are on the left screen, 
I suggest making them darker so that the user can see the differences right away.

Screenshot_20230123-123340

**Comment by srujner** [comment]
**Tested with Success!**

Verified on Android 12, 13

**Verified Cases:**
- New autoincrement mode;
- Overall regression check on the Counter App;
- Change Language; 
- Rotate the screen, minimize the app;

**Comment by dbemke** [comment]
**Tested with Success!**

Verified on Android 8.1, 10

**Comment by grzesiek2010** [comment]
> I suggest making them darker so that the user can see the differences right away.

I added that change:
Screenshot_1674563952

**Review by lognaturel — APPROVED** [review]

**Pull request merged**